pkg/virt-operator/resource/apply/reconcile_test.go

```diff
@@ -552,14 +552,21 @@ var _ = Describe("Apply", func() {
 				Expect(preferredSchedulingTerm.Weight).To(Equal(int32(100)))
 				Expect(preferredSchedulingTerm.Preference.MatchFields).To(BeEmpty())
 				Expect(preferredSchedulingTerm.Preference.MatchExpressions).To(HaveLen(1))
-
-				matchExpression := preferredSchedulingTerm.Preference.MatchExpressions[0]
-				Expect(matchExpression.Key).To(Equal(workerLabel))
-				Expect(matchExpression.Operator).To(Equal(corev1.NodeSelectorOpDoesNotExist))
 			},
 				Entry("nil component config", nil),
 				Entry("nil node placement", &v1.ComponentConfig{}),
 			)
+
+			It("should not add requirement scheduling to control-plane nodes if the user already provided a node placement", func() {
```

Comment:
I think this case can be merged with the `DescribeTable` above.

Reply:
TBH I'm not sure it'll make it easier to read or understand. The describe table deals with the case of adding the requirements (i.e. a positive test). Adding negative tests on the same table seems uncohesive to me.

Instead, I've added a dedicated context for negative tests. The tests there are completely straight-forward almost one-liner tests, and I'm not sure merging them into a table would do more good than damage

Reply:
Yeah, I guess it's a matter of preference :)
I just tend to think that super obvious and short test cases are easier to read with a dedicated test. But I see your point